<launch>
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="false"/>
  <arg name="multi_robot_name" default=""/>

  <!-- move_base -->
	<group ns="$(arg multi_robot_name)">
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <rosparam file="$(find motion_planning)/param/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" /> <!-- -->
    <rosparam file="$(find motion_planning)/param/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" /> <!-- -->
    <rosparam file="$(find motion_planning)/param/local_costmap_params.yaml" command="load" /> <!-- -->
    <rosparam file="$(find motion_planning)/param/global_costmap_params.yaml" command="load" /> <!-- -->
    <rosparam file="$(find motion_planning)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find motion_planning)/param/dwa_local_planner_params_$(arg model).yaml" command="load" /> <!-- -->

		<param name="global_costmap/robot_base_frame" value="$(arg multi_robot_name)/base_footprint"/> <!-- -->
		<param name="local_costmap/robot_base_frame" value="$(arg multi_robot_name)/base_footprint"/> <!-- -->

		<param name="local_costmap/global_frame" value="$(arg multi_robot_name)/odom"/> <!-- -->
		<param name="global_costmap/global_frame" value="$(arg multi_robot_name)/map"/> <!-- -->

		<param name="global_costmap/scan" value="{sensor_frame: $(arg multi_robot_name)/base_scan, data_type: LaserScan, topic: $(arg multi_robot_name)/scan, marking: true, clearing: true}"/> <!-- -->
		<param name="local_costmap/scan" value="{sensor_frame: $(arg multi_robot_name)/base_scan, data_type: LaserScan, topic: $(arg multi_robot_name)/scan, marking: true, clearing: true}"/> <!-- -->

    <param name="global_costmap/observation_source" value="$(arg multi_robot_name)/scan"/> <!-- -->
    <param name="local_costmap/observation_source" value="$(arg multi_robot_name)/scan"/> <!-- -->

		<!--remap from="scan" to="$(arg multi_robot_name)/scan"/-->

    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
  </node>
	</group>
</launch>
